hazelcast/src/main/java/com/hazelcast/map/impl/recordstore/AbstractEvictableRecordStore.java

```diff
@@ -126,7 +126,7 @@ public void evictExpiredEntryAndPublishExpiryEvent(Data key,
                                                        ExpiryReason expiryReason,
                                                        boolean backup) {
         Object value = evict(key, backup);
-        if (!backup) {
+        if (value != null && !backup) {
```

Comment:
Under what circumstances can `value` be `null`?

Reply:
when no key exists in map, evict returns null. In expiry-system, we have a cached iterator and it can produce this kind of anomalies.

Reply:
Thanks, it makes sense then :+1: